examples/by_runtime_api_module/array_management.cu

```diff
@@ -0,0 +1,98 @@
+#include <cuda/api_wrappers.hpp>
+
+#include <iostream>
+
+__global__ void from_texture_to_ptr_kernel3D(cudaTextureObject_t texture_obj, float* ptr, size_t w, size_t h, size_t d) {
+
+	const auto gtidx = threadIdx.x + blockIdx.x * blockDim.x;
+	const auto gtidy = threadIdx.y + blockIdx.y * blockDim.y;
+	const auto gtidz = threadIdx.z + blockIdx.z * blockDim.z;
+	const auto gtid = gtidx + gtidy * w + gtidz * w * h;
+
+	if (gtidx < w && gtidy < h && gtidz < d) {
+		ptr[gtid] = tex3D<float>(texture_obj, gtidx, gtidy, gtidz);
+	}
+}
+
+__global__ void from_texture_to_ptr_kernel2D(cudaTextureObject_t texture_obj, float* ptr, size_t w, size_t h) {
+
+	const auto gtidx = threadIdx.x + blockIdx.x * blockDim.x;
+	const auto gtidy = threadIdx.y + blockIdx.y * blockDim.y;
+	const auto gtid = gtidx + gtidy * w;
+
+	if (gtidx < w && gtidy < h) {
+		auto x = tex2D<float>(texture_obj, gtidx, gtidy);
+		printf("thread %u %u, reading value %f, and writing to index %u\n", gtidx, gtidy, x, gtid);
+		ptr[gtid] = x;
+	}
+}
+
+
+int main() {
+
+	auto device = cuda::device::current::get();
+
+	//
+	// 3D
+	//
+	constexpr size_t w = 3;
+	constexpr size_t h = 3;
+	constexpr size_t d = 3;
+	cuda::array::dimensions_t<3> dims = {w, h, d};
+
+	cuda::array::array_t<float, 3> arr0(device, dims);
+	auto ptr_in0 = cuda::memory::managed::make_unique<float[]>(arr0.size());
```

Comment:
Do you know some libraries, which contain multi dimensional containers? Maybe we can compare how they chose their naming. Boost multi array uses `.size()` for `boostmultiarray.shape()[0]`, which I think is confusing. But they have a function `boostmultiarray.shape()` to return the dimensions of the array. I would agree with `auto dims = 
 cudaarray.dims()`, but I am not sure about wether `dims.volume()` or `cudaarray.total_size()` is a good choice. In my own code I write `cudaarray.dims().reduce_mult()` because `.dims()` returns some kind of array type and we calculate a multiplicative reduction on that, but I am also not sure if this is a good choice. Would be really good to have a look at some other modern libraries. The problem is that e.g. boost::array and std::array do not have any kind of multiplicative reduction function implemented.

Reply:
Well, I guess there's gsl::multispan , and there's eigen. We can look at those.